src/main/java/org/myrobotlab/i2c/I2CFactory.java

```diff
@@ -42,7 +51,14 @@ public static I2CBus getInstance(int busNumber) throws IOException {
       // TODO: fix this!!!
       //log.warn("This probable doesn't work for ARM / RasPI now.. update the code!");
       //return I2CBusImpl.getBus(busNumber);
-      return null;
+      //return null;
+      try {
+        return provider.getBus(busNumber, DEFAULT_LOCKAQUIRE_TIMEOUT, DEFAULT_LOCKAQUIRE_TIMEOUT_UNITS);
+      } catch (UnsupportedBusNumberException e) {
+        // TODO Auto-generated catch block
+        e.printStackTrace();

```

Comment:
Generated code doesn't always do we want ...
Please change to a logger, because std output doesn't necessarily go to the log appender - also it can not be controlled, filtered or assigned to debug, info, warn or error

catch(UnsupportedBusNumberException e) {
       log.error("getBus threw", e);

The above will send it to the current logger, and also categorize it as error - and if the logger is created as all other loggers in mrl - it will show that this class did the logging, allowing quick identification and more filtering options.

Reply:
oh, u right, it was laziness will fix it